src/gurt/telemetry.c

```diff
@@ -733,6 +733,8 @@ destroy_shmem_with_key(key_t key)
 	header->sh_deleted = 1;
 	head = &header->sh_subregions;
 	for (cur = conv_ptr(header, head->next); cur != head; cur = conv_ptr(header, cur->next)) {
+		if (cur == NULL)
```

Comment:
maybe this should be assert?

Reply:
The returned value of conv_ptr could be either NULL or non-NULL. It seems that it is checked for NULL in some other places.

Reply:
In theory the shmem headers/pointers *could* be malformed. I think breaking out of the loop is the right thing to do.